### DISCUSSION TITLE
Installing hotfixes

### DISCUSSION THREAD

**Discussion by petiepooo** [Security-Onion-Solutions/securityonion#7974] [Unsupported Versions] [answered]
Is there a way to install just a hotfix when there is a new base version available but we're not ready to upgrade yet?

The only documentation I've found says that soup will simply install the most current version without giving you any choice.

**Comment by dougburks** [comment, marked as answer]
soup always installs the latest version and, if available, the latest hotfix for that version.

**Reply by petiepooo** [reply]
So let's say I have 2.3.100 (no hotfixes) installed, and 2.3.110 comes out.  I'd like to follow your advice about custom templates and hold off installing 2.3.110, but would like to install any available hotfixes for 2.3.100.  Your official answer is that is not possible?

I'd like to point out that you recognize the value of doing QA testing before upgrades, @dougburks, as you actively pin the salt and docker versions for that reason.  To deny us that option and insist that the only option we have is to upgrade to the latest version or nothing is hypocritical, IMO.  We have QA to do on our toolchain too, and cannot always upgrade to the latest SO2 version at a moment's notice.

For instance, when upgrading to 2.3.100, the method for pivoting to a PCAP changed.  It seems a small thing, but our level-1 analysts revolted when they realized they now had to open a separate page and pull down another menu for a feature they use all the time.  I need time to discover and document changes like that before rolling out the next version.

Please give us the ability to choose to what level we wish to upgrade when running soup.  At the very least, give us the option to either upgrade to the latest or just install hotfixes for the version we are currently running.

**Reply by dougburks** [reply]
The official answer is "soup always installs the latest version and, if available, the latest hotfix for that version."

The unofficial answer is: if you take a look at the code inside soup, you might find an undocumented and UNSUPPORTED option that might help you achieve your goal.

Finally, replies like the one above come across as adversarial, condescending, and accusatory. Please review the etiquette guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 before posting again.

**Reply by petiepooo** [reply]
My apologies.  I see now that hypocritical is a more adversarial term than I should have used  Perhaps short-sighted would be a better fit, as I think that many development efforts focus more on the things right in front of them, often to the detriment of understanding where their piece fits into the larger picture.  The point I was inelegantly trying to make is that upgrading to the latest version can introduce issues of its own, and choosing the version to install/upgrade is an important ability of a robust solution.

I believe I see the eluded way to specify a "branch", and appreciate your tip.  Please consider documenting it, even if it's with a disclaimer saying SOS does not recommend or support its use.  It would have reduced my frustration level, to be sure.

**Reply by dougburks** [reply]
Now you're accusing us of being `short-sighted`? That still comes across as condescending.

This is your FINAL warning. If you cannot follow the etiquette guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720, then your account will be banned.

**Comment by TotieBash** [comment]
I know on "airgap" configuration you can provide the iso location so theoretically you can provide the iso update that you choose.  
https://docs.securityonion.net/en/2.3/soup.html?highlight=update%20airgap#airgap
